src/sshfs_mount/sshfs_mount.cpp

```diff
@@ -50,6 +50,25 @@ auto run_cmd(mp::SSHSession& session, std::string&& cmd)
     return run_cmd(session, std::forward<std::string>(cmd), error_handler);
 }
 
+// Run a command returning a string which can end in space on a given session and reads the result. It adds a
+// non-space character (actually, a '-') to signal the end of the resulting string, and later removes the trailing
+// spaces and the special character to gather the contents of the return string including trailing spaces of the
+// command output.
```

Comment:
This is catering for a very extreme edge case. IMO it adds complexity to handle something, which most often will be a mistake (trailing space in home folder).

@ricab @townsend2010 WDYT?

I wonder if the right thing to do would be to _not_ trim the string (other than removing any expected newlines)?

I mean, we're in control of the commands, and should be able to assume the behaviour of them to not return extraneous whitespace?

Reply:
Hmm, I am uncertain. It is not very complex, and if valid homes can have trailing spaces, I don't think we should remove them. But I also don't understand why we would remove *any* space. 

I would like to know what @luis4a0 has to say about it. What is the motivation for trimming in the first place? This seems to deal with output with an unknown amount of space padding, but what is the actual use case for that? Why would we be getting unexpected spaces from the commands we issue? I mean, when we parse output, we need to be able to assume some format, right? So why aren't spaces accounted for in that format? Some quirk of different shells? Or just being careful, just in case?

Also, please remind me, why parse bits of the path ourselves? Why not just leave the whole thing to the shell? IOW, isn't this just a particular case of the earlier issue, where we could just do `readlink -f input`? It requires the instance to be running, but so does this, right?

Reply:
> But I also don't understand why we would remove any space.

It's kinda the default when dealing with shell output, because you will always get a trailing newline that's not part of the _interesting_ output. But yeah that's what I meant by suggesting to not trim it in the first place, or rather - only trimming the "expected" trailing character.

We (`multipass exec`) actually seem to be dealing better with this than `ssh`:
```console
$ echo "foo " | hexdump 
0000000 6f66 206f 000a                         
0000005
$ echo -n "foo " | hexdump
0000000 6f66 206f                              
0000004
$ ssh localhost echo "foo " | hexdump  
0000000 6f66 0a6f                              
0000004
$ ssh localhost echo -n "foo " | hexdump  # notice SSH trims the space in the echoed string
0000000 6f66 006f                              
0000003
$ mp exec primary -- echo "foo " | hexdump
0000000 6f66 206f 000a                         
0000005
$ mp exec primary -- echo -n "foo " | hexdump
0000000 6f66 206f                              
0000004
```

> Also, please remind me, why parse bits of the path ourselves?

The only other way to expand `~` would be evaluating it in the shell… While it should generally be OK, since users "have root" on the instances anyway, so they don't _need_ to do "BASH injection" attacks, it felt cleaner to attack this through `getent`. We can error out a bit better.
The adverse effect is that `multipass mount . instance:~1000/path` does work, when it shouldn't…

Reply:
My intention here was to make reading output from commands more robust. But it's true that it adds complexity to the implementation for very small improvement, for a corner case, at the cost of adding code which is not so easy to maintain. I agree that forgetting about this improvement will simplify things a lot.

Reply:
I agree this adds quite a bit of complexity for a very small corner case.  I'm +1 on removing this.